src/integer_mod_q/mat_zq/properties.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2023 Phil Milewski
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality about properties of [`MatZq`] instances.
+
+use super::MatZq;
+use flint_sys::{
+    fmpz_mat::fmpz_mat_is_one,
+    fmpz_mod_mat::{fmpz_mod_mat_is_square, fmpz_mod_mat_is_zero},
+};
+
+impl MatZq {
+    /// Checks if a [`MatZq`] is the identity matrix.
+    ///
+    /// Returns true if every diagonal entry of the upper square matrix is `1`
+    ///  and all other entries are `0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatZq::from_str("[[1, 0],[0, 1]] mod 17").unwrap();
+    /// assert!(value.is_identity())
+    /// ```
```

Comment:
I like the test case you added for `MatQ::is_identity` to make it more clear that non-square matrices are allowed.
```suggestion
    /// ```
    ///
    /// ```
    /// use qfall_math::integer_mod_q::MatZq;
    /// use std::str::FromStr;
    ///
    /// let value = MatZq::from_str("[[1, 0],[0, 1], [0, 0]] mod 17").unwrap();
    /// assert!(value.is_identity());
    /// ```
```